### PR TITLE
Add docs for proxy-set-header annotation in Ingress

### PR DESCRIPTION
**Pull request by AlexFenlon**
### Proposed changes

Docs for proxy-set-headers annotation for Ingress, will be adding examples & tutorial soon as a separate pr.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: examples/ingress-resources/proxy-set-headers/README.md [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+  proxy_set_header X-Forwarded-Tea "greenTea";
+  
+  ...
+...
+
+```
+
+### Example 2: Minion Override Master Header
+
+In this example, a header is in both a master and a minion, in the minion it has a different value and therefore overrides master.
```
Comment (Jcahilltorre) [inline examples/ingress-resources/proxy-set-headers/README.md:223]:
```suggestion
In this example, a header is in both a master and a minion; in the minion it has a different value and therefore overrides master.
```

File: examples/ingress-resources/proxy-set-headers/README.md [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+
+## Proxy-Set-Headers Annotation In Mergeable Ingress Type
+
+When using the proxy-set-headers annotation in minions, if there are conflicting headers with different values between master and minion, the minion's value will override the master's.
+
+If different headers are specified in both master and minions, both sets of annotations will be applied.
+
+### Example 1: Headers in both Master and Minions
+
+In this example, we add a custom header and value to each minion with defaults for Master
```
Comment (Jcahilltorre) [inline examples/ingress-resources/proxy-set-headers/README.md:124]:
```suggestion
In this example, we add a custom header and value to each minion with defaults for master:
```

File: examples/ingress-resources/proxy-set-headers/README.md [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+...
+
+  proxy_set_header X-Forwarded-ABC $http_x_forwarded_abc;
+
+...
+```
+
+### Example 2: Setting Multiple Custom Headers With Default Values
+
+In this example, the ``nginx.org/proxy-set-headers`` annotation is used to set multiple custom headers: `X-Forwarded-ABC` and `ABC2`.
```
Comment (Jcahilltorre) [inline examples/ingress-resources/proxy-set-headers/README.md:53]:
```suggestion
In this example, the ``nginx.org/proxy-set-headers`` annotation is used to set multiple custom headers: `X-Forwarded-ABC` and `ABC2`:
```

File: examples/ingress-resources/proxy-set-headers/README.md [hunk 4]
```diff
@@ -0,0 +1,410 @@
...
+- **Multiple Headers**: Set multiple custom headers by separating their names with commas.
+- **Custom Header Values**: Set custom values for headers by using the format `"Header-Name: Value"`.
+
+When using the ``proxy-set-headers`` annotation, the specified headers will be added to the outgoing requests proxied by NGINX.
+
+## Proxy-Set-Headers Annotation In Standard Ingress Type
+
+### Example 1: Setting a Single Custom Header With Default Value
+
+In the following example, the ``nginx.org/proxy-set-headers`` annotation is used to set a single custom header named `X-Forwarded-ABC`.
```
Comment (Jcahilltorre) [inline examples/ingress-resources/proxy-set-headers/README.md:21]:
```suggestion
In the following example, the ``nginx.org/proxy-set-headers`` annotation is used to set a single custom header named `X-Forwarded-ABC`:
```

**Review by Jcahilltorre — APPROVED** [review]

File: examples/ingress-resources/proxy-set-headers/README.md [hunk 5]
```diff
@@ -0,0 +1,410 @@
...
+  proxy_set_header X-Forwarded-Minion "tea";
+  ...
+
+...
+
+```
+
+### Example 3: No Annotation in Minion
+
+In this example, we add the annotation in master to add a custom header and value to a minion without the annotation
```
Comment (ADubhlaoich) [inline examples/ingress-resources/proxy-set-headers/README.md:320]:
```suggestion
In this example, we use the annotation in master to add a custom header and value to a minion without an annotation:
```

File: examples/ingress-resources/proxy-set-headers/README.md [hunk 6]
```diff
@@ -0,0 +1,410 @@
...
+        backend:
+          service:
+            name: tea-svc
+            port:
+              number: 80
+```
+
+Corresponding NGINX config file snippet:
+
+```bash
```
Comment (ADubhlaoich) [inline examples/ingress-resources/proxy-set-headers/README.md:393]:
```suggestion
```shell
```

File: examples/ingress-resources/proxy-set-headers/README.md [hunk 7]
```diff
@@ -0,0 +1,410 @@
...
```
Comment (ADubhlaoich) [inline examples/ingress-resources/proxy-set-headers/README.md:297]:
```suggestion
```shell
```

File: examples/ingress-resources/proxy-set-headers/README.md [hunk 8]
```diff
@@ -0,0 +1,410 @@
...
+      paths:
+      - path: /
+        backend:
+          serviceName: example-service
+          servicePort: 80
+```
+
+Corresponding NGINX config file snippet:
+
+```bash
```
Comment (ADubhlaoich) [inline examples/ingress-resources/proxy-set-headers/README.md:75]:
```suggestion
```shell
```

File: examples/ingress-resources/proxy-set-headers/README.md [hunk 9]
```diff
@@ -0,0 +1,410 @@
+# Support for custom proxy headers
+
+NGINX and NGINX Plus support customization of proxy headers through annotations in Ingress.
```
Comment (ADubhlaoich) [inline examples/ingress-resources/proxy-set-headers/README.md:3]:
```suggestion
You can customize proxy headers for NGINX and NGINX Plus using Ingress annotations.
```

**Review by ADubhlaoich — APPROVED** [review]
LGTM, made a few suggestions.

File: examples/ingress-resources/proxy-set-headers/README.md [hunk 10]
```diff
@@ -0,0 +1,410 @@
...
+Content of `cafe-master.yaml`:
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: cafe-ingress-master
+  annotations:
+    nginx.org/mergeable-ingress-type: "master"
+    nginx.org/proxy-set-headers: "X-Forwarded-ABC master"
```
Comment (j1m-ryan) [inline examples/ingress-resources/proxy-set-headers/README.md:234]:
```suggestion
    nginx.org/proxy-set-headers: "X-Forwarded-ABC: master"
```

File: examples/ingress-resources/proxy-set-headers/README.md [hunk 11]
```diff
@@ -0,0 +1,410 @@
...
+Content of `coffee-minion.yaml`:
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: cafe-ingress-coffee-minion
+  annotations:
+    nginx.org/mergeable-ingress-type: "minion"
+    nginx.org/proxy-set-headers: "X-Forwarded-Coffee cappuccino,X-Forwarded-ABC coffee"
```
Comment (j1m-ryan) [inline examples/ingress-resources/proxy-set-headers/README.md:254]:
```suggestion
    nginx.org/proxy-set-headers: "X-Forwarded-Coffee: cappuccino,X-Forwarded-ABC: coffee"
```

File: examples/ingress-resources/proxy-set-headers/README.md [hunk 12]
```diff
@@ -0,0 +1,410 @@
...
+Content of `coffee-minion.yaml`:
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: cafe-ingress-coffee-minion
+  annotations:
+    nginx.org/mergeable-ingress-type: "minion"
+    nginx.org/proxy-set-headers: "X-Forwarded-Coffee cappuccino"
```
Comment (j1m-ryan) [inline examples/ingress-resources/proxy-set-headers/README.md:351]:
```suggestion
    nginx.org/proxy-set-headers: "X-Forwarded-Coffee: cappuccino"
```

**Review by jjngx — APPROVED** [review]
👍🏻

**Pull request merged**